core-bundle/src/EventListener/Menu/BackendFavoritesListener.php

```diff
@@ -0,0 +1,160 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CoreBundle\EventListener\Menu;
+
+use Contao\BackendUser;
+use Contao\CoreBundle\Event\MenuEvent;
+use Doctrine\DBAL\Connection;
+use Knp\Menu\FactoryInterface;
+use Knp\Menu\ItemInterface;
+use Knp\Menu\Util\MenuManipulator;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\RequestStack;
+use Symfony\Component\Routing\RouterInterface;
+use Symfony\Component\Security\Core\Security;
+use Symfony\Contracts\Translation\TranslatorInterface;
+
+/**
+ * @internal
+ */
+class BackendFavoritesListener
+{
+    public function __construct(
+        private Security $security,
+        private RouterInterface $router,
+        private RequestStack $requestStack,
+        private Connection $connection,
+        private TranslatorInterface $translator,
+    ) {
+    }
+
+    public function __invoke(MenuEvent $event): void
+    {
+        $user = $this->security->getUser();
+
+        if (!$user instanceof BackendUser) {
+            return;
+        }
+
+        $name = $event->getTree()->getName();
+
+        if ('mainMenu' !== $name) {
+            return;
+        }
+
+        $this->buildFavoritesMenu($event, $user);
+    }
+
+    private function buildFavoritesMenu(MenuEvent $event, BackendUser $user): void
+    {
+        if (!$request = $this->requestStack->getCurrentRequest()) {
+            throw new \RuntimeException('The request stack did not contain a request');
```

Comment:
```suggestion
            return;
```

Why throwing an exception? You just cannot contribute to the menu but 🤷‍♂️

Reply:
Changed in c4ea87858276c033101994efaf2c0bca9999ed97.